snr.py

```diff
@@ -0,0 +1,287 @@
+import numpy as np
+import astropy.units as u
+from astropy.tests.helper import assert_quantity_allclose
+from scipy.optimize import fsolve
+
+
+@u.quantity_input(counts=u.electron,
+                  npix=u.pixel,
+                  n_background=u.pixel,
+                  background=u.electron / u.pixel,
+                  darkcurrent=u.electron / u.pixel,
+                  readnoise=u.electron / u.pixel,
+                  gain=u.electron / u.adu,
+                  ad_err=u.adu / u.pixel)
+def howell_snr(counts,
+               npix=1 * u.pixel,
+               n_background=np.inf * u.pixel,
+               background=0 * u.electron / u.pixel,
+               darkcurrent=0 * u.electron / u.pixel,
+               readnoise=0 * u.electron / u.pixel,
+               gain=1 * u.electron / u.adu,
+               ad_err=np.sqrt(0.289) * u.adu / u.pixel):
+    """
+    A function to calculate the idealized theoretical signal to noise ratio
+    (SNR) of an astronomical observation with a given number of counts. This
+    expression is taken from "Handbook of CCD Astronomy", Steve Howell, 2000,
+    pg 55).
+
+    Parameters
+    ----------
+    counts : `~astropy.units.Quantity`
+        Total number of counts in an arbitrary exposure time with units of
+        electrons.
+    npix : `~astropy.units.Quantity`, optional
+        Number of pixels under consideration for the signal with units of
+        pixels. Default is 1 * astropy.units.pixel.
+    n_background : `~astropy.units.Quantity`, optional
+        Number of pixels used in the background estimation with units of
+        pixels. Default is set to np.inf * astropy.units.pixel
+        such that there is no contribution of error due to background
+        estimation. This assumes that n_background will be >> npix.
+    background : `~astropy.units.Quantity`, optional
+        Total photons per pixel due to the background/sky with units of
+        electrons/pixel.
+        Default is 0 * astropy.units.electron / astropy.units.pixel.
+    darkcurrent : `~astropy.units.Quantity`, optional
+        Total electrons per pixel due to the dark current with units
+        of electrons/pixel.
+        Default is 0 * astropy.units.electron / astropy.units.pixel.
+    readnoise : `~astropy.units.Quantity`, optional
+        Electrons per pixel from the read noise with units of electrons/pixel.
+        Default is 0 * astropy.units.electron / astropy.units.pixel.
+    gain : `~astropy.units.Quantity`, optional
+        Gain of the CCD with units of electrons/ADU. Default is
+        1 * astropy.units.electron / astropy.units.adu such that the
+        contribution to the error due to the gain is assumed to be small.
+    ad_err : `~astropy.units.Quantity`, optional
+        An estimate of the 1 sigma error within the A/D converter with units of
+        adu/pixel. Default is set to
+        sqrt(0.289) * astropy.units.adu / astropy.units.pixel
+        (Merline & Howell, 1995).
```

Comment:
Would be nice to cite this according to ApJ standard: https://journals.aas.org/references/

Example on how to cite in a docstring:
https://github.com/astropy/astropy/blob/7e751f16ec3bad239aea30790e4875226216a7d3/astropy/timeseries/periodograms/lombscargle/core.py#L94-L100

This comment also applies to other similar occurrences.